### PR TITLE
Sidekiq and Mailgun

### PR DESCRIPTION
**Pull request by JoaquinSeita**
## Summary of what has been implemented

- [x] Sidekiq workers.
- [x] Mailer and mailgun working together.
- [x] Confirmation email contains the requested data of the rent.

## Screenshots

<h3>Tests results and coverage</h3>

Captura de pantalla de 2021-11-04 17-19-46

<h3>A rent created</h3>

Captura de pantalla de 2021-11-04 15-11-27

<h3>The confirmation email that the user receives</h3>

Captura de pantalla de 2021-11-04 15-10-27

## TRELLO Card
[Send email to the user when a rent is created](https://trello.com/c/THqlWhdO/7-mandar-email-al-usuario-cuando-se-crea-un-rent)

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'postmaster@sandboxd23676cf8ca74d14b04278ed1d309c95.mailgun.org'
```
Comment (lucasenriquez) [inline app/mailers/application_mailer.rb:2]: This variable should be in your env file

File: app/views/layouts/mailer.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
```
Comment (lucasenriquez) [inline app/views/layouts/mailer.html.erb:6]: You should delete unnecesary comments and unused code

File: app/workers/hard_worker.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class HardWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: false
+
+  def perform(rent_id)
+    @rent = Rent.find(rent_id)
```
Comment (lucasenriquez) [inline app/workers/hard_worker.rb:6]: Why do you declare the variable as global when you only use it here?

File: config/environments/development.rb [hunk 4]
```diff
@@ -15,9 +15,18 @@
...
+  config.action_mailer.raise_delivery_errors = true
   config.action_mailer.perform_caching = false
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    :domain => ENV['MAILER_DOMAIN'],
+    :password => ENV['MAILER_PASSWORD'],
+    :user_name => ENV['MAILER_USERNAME'],
+    :address => "smtp.mailgun.org",
+    :port => 587,
```
Comment (lucasenriquez) [inline config/environments/development.rb:27]: This should be in your env file

File: config/initializers/sidekiq.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Sidekiq.configure_server do |config|
+    config.redis = { url: ENV.fetch('REDIS_URL_SIDEKIQ', 'redis://localhost:6379/1') }
+end
+
+Sidekiq.configure_client do |config|
+    config.redis = { url: ENV.fetch('REDIS_URL_SIDEKIQ', 'redis://localhost:6379/1') }
+end
```
Comment (lucasenriquez) [inline config/initializers/sidekiq.rb:7]: Add a newline at the endd of the file

File: spec/mailers/rents_mailer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  let(:user_mail) { create(:user, email: 'juan@gmail.com') }
+  let(:rent) { create(:rent, user_id: user_mail.id, book_id: book.id) }
+
+  describe 'rent_confirmation' do
+    let(:mail) { RentsMailer.rent_confirmation(rent) }
+
+    it 'check the headers' do
+      expect(mail.subject).to eq('[NO-REPLY] Details about the book rental!')
+      expect(mail.from).to eq(['postmaster@sandboxd23676cf8ca74d14b04278ed1d309c95.mailgun.org'])
```
Comment (lucasenriquez) [inline spec/mailers/rents_mailer_spec.rb:14]: This kind of data should be in your env file

**Review by lucasenriquez — CHANGES_REQUESTED** [review]
I saw that you are not using the secrets file. Why is that?

File: config/secrets.ci.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+mailer_domain: <%= ENV['MAILER_DOMAIN'] %>
+mailer_password: <%= ENV['MAILER_PASSWORD'] %>
+mailer_username: <%= ENV['MAILER_USERNAME'] %>
+mailer_address: <%= ENV['MAILER_ADDRESS'] %>
+mailer_port: <%= ENV['MAILER_PORT'] %>
+mailer_authentication: <%= ENV['MAILER_AUTHENTICATION'] %>
```
Comment (lucasenriquez) [inline config/secrets.ci.yml:6]: Add a newline at the end of the file

**Review by lucasenriquez — CHANGES_REQUESTED** [review]

File: app/mailers/application_mailer.rb (hunk 1 above)
Comment (JoaquinSeita) [reply app/mailers/application_mailer.rb]: Now stored in encrypted credentials

File: config/environments/development.rb (hunk 4 above)
Comment (JoaquinSeita) [reply config/environments/development.rb]: Now it is stored in the encrypted credentials too

File: spec/mailers/rents_mailer_spec.rb (hunk 6 above)
Comment (JoaquinSeita) [reply spec/mailers/rents_mailer_spec.rb]: Okay! Now a variable from the credentials is called

File: config/secrets.ci.yml (hunk 7 above)
Comment (JoaquinSeita) [reply config/secrets.ci.yml]: Thanks! add it

**Review by lucasenriquez — APPROVED** [review]

**Review by maxwidergy — APPROVED** [review]
ok

**Pull request merged**